sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -166,6 +197,25 @@ class FileStreamSource(
         // implies "sourceOptions.latestFirst = true" which we want to refresh the list per batch
         (newFiles.take(files.maxFiles()), null)
 
+      case files: ReadMaxBytes if !sourceOptions.latestFirst =>
+        // we can cache and reuse remaining fetched list of files in further batches
+        val (FilesSplit(bFiles, _), FilesSplit(usFiles, rSize)) =
+          takeFilesUntilMax(newFiles, files.maxBytes())
+        if (rSize < (files.maxBytes() * DISCARD_UNSEEN_FILES_RATIO).toLong) {
```

Comment:
I think `DISCARD_UNSEEN_FILES_RATIO` is designed for max files `ReadMaxFiles`. I'm not sure if it makes sense to directly apply it for `ReadMaxBytes`.

Reply:
Agree, should we create a separate one or maybe just rename to something like `DISCARD_UNSEEN_INPUT_RATIO`?

Reply:
I think we can rename it to `DISCARD_UNSEEN_INPUT_RATIO` for now.